Add initial docs for Go gRPC. (#2)
smallstep/practicalzt: src/docs/go-grpc/sections/02-try/02-server.mdx

> We're going to create a simple gRPC server in Go that loads an X.509 certificate
> and its key to authenticate to clients using just server authentication.

Reviewer: ```suggestion
We're going to create a simple gRPC server in Go that loads an X.509 certificate and its key,
so that clients can connect using TLS server authentication.
```

First, let's create our Go module:

```shell
mkdir helloworld
cd helloworld
go mod init example.com/helloworld
```

> Now we're going to implement this
> [helloworld.proto](https://github.com/grpc/grpc-go/blob/master/examples/helloworld/helloworld/helloworld.proto)
> in a file named `main.go`. The code loads the X.509 certificate and its key and
> runs a gRPC server:
> 

Reviewer: ```suggestion
Now we're going to implement the [helloworld.proto](https://github.com/grpc/grpc-go/blob/master/examples/helloworld/helloworld/helloworld.proto) `Greeting` service in a file named `main.go`.
The code loads the X.509 certificate and its key
and starts a gRPC server:

```

```go
package main

import (
	"context"
	"crypto/tls"
	"log"
	"net"

	"google.golang.org/grpc"
	"google.golang.org/grpc/credentials"
	"google.golang.org/grpc/examples/helloworld/helloworld"
	"google.golang.org/grpc/reflection"
)

type helloServer struct {
	helloworld.UnimplementedGreeterServer
}

func (*helloServer) SayHello(ctx context.Context, in *helloworld.HelloRequest) (*helloworld.HelloReply, error) {
	log.Printf("method=SayHello name=%s", in.GetName())
	return &helloworld.HelloReply{Message: "Hello " + in.GetName()}, nil
}

func main() {
	cert, err := tls.LoadX509KeyPair("server.crt", "server.key")
	if err != nil {
		log.Fatal(err)
	}

	opts := []grpc.ServerOption{
		grpc.Creds(credentials.NewServerTLSFromCert(&cert)),
	}

	lis, err := net.Listen("tcp", ":443")
	if err != nil {
		log.Fatalf("failed to listen: %v", err)
	}

	srv := grpc.NewServer(opts...)
	helloworld.RegisterGreeterServer(srv, &helloServer{})
	reflection.Register(srv)

	log.Printf("server listening at %v", lis.Addr())
	if err := srv.Serve(lis); err != nil {
		log.Fatalf("failed to serve: %v", err)
	}
}
```

Now let's download the dependencies and compile it:

> ```shell
> go mod tidy
> go build main.go
> ```

Reviewer 2: ```suggestion
<CodeBlock language="shell" copyText="go mod tidy && go build main.go">
{`go mod tidy
go build main.go`}
</CodeBlock>
```
